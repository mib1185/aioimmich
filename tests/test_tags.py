"""Tests for aioimmich."""

from __future__ import annotations

from syrupy.assertion import SnapshotAssertion


async def test_get_all_tags(mock_immich_with_data, snapshot: SnapshotAssertion):
    """Test async_get_all_tags."""
    api = await mock_immich_with_data()
    tags = await api.tags.async_get_all_tags()

    assert len(tags) == 3
    assert tags == snapshot
