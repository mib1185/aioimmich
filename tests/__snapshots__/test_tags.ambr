# serializer version: 1
# name: test_get_all_tags
  list([
    ImmichTag(created_at=datetime.datetime(2025, 5, 12, 20, 1, 24, 330000, tzinfo=datetime.timezone.utc), name='My 30th birthday', tag_id='14ce3af3-67be-41c6-b77c-b25abddaf546', updated_at=datetime.datetime(2025, 5, 12, 20, 1, 24, 412000, tzinfo=datetime.timezone.utc), value='My 30th birthday', color=None, parent_id=None),
    ImmichTag(created_at=datetime.datetime(2025, 5, 12, 20, 0, 45, 220000, tzinfo=datetime.timezone.utc), name='Halloween', tag_id='67301cb8-cb73-4e8a-99e9-475cb3f7e7b5', updated_at=datetime.datetime(2025, 5, 12, 20, 0, 47, 224000, tzinfo=datetime.timezone.utc), value='Halloween', color=None, parent_id=None),
    ImmichTag(created_at=datetime.datetime(2025, 5, 12, 20, 0, 49, 967000, tzinfo=datetime.timezone.utc), name='Holidays', tag_id='69bd487f-dc1e-4420-94c6-656f0515773d', updated_at=datetime.datetime(2025, 5, 12, 20, 0, 55, 575000, tzinfo=datetime.timezone.utc), value='Holidays', color=None, parent_id=None),
  ])
# ---
